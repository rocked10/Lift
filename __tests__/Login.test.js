import 'react-native';
import React from 'react';
import { Alert } from 'react-native';
import renderer from 'react-test-renderer';
import Login from '../screens/login';
import { render, fireEvent, waitFor, act } from "@testing-library/react-native";

describe('Testing Login Screen', () => {
    const handleLogin = jest.fn();
    const wrapper = renderer.create(<Login userDetails={handleLogin} />);

    it('Should render', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it ("Renders elements", () => {    
        const { getByTestId, getAllByText, getByPlaceholderText, } = render(<Login userDetails={handleLogin} />);
        getByTestId('Lift Logo');
        getByPlaceholderText("Email...");
        getByPlaceholderText("Password...");
        expect(getAllByText("LOGIN").length).toBe(1);
        expect(getAllByText("SIGNUP").length).toBe(2);
    });

    it("Workout screen does not show if user tries to login without email and password", async () => {
        const { queryByTestId, getByTestId} = render(<Login userDetails={handleLogin} />);
        await waitFor(() => fireEvent.press(getByTestId("Login Button")));
        expect(queryByTestId('Add Workout')).toBeNull();
    })
});

