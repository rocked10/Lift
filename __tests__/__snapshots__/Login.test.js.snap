// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Login Screen Should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "borderRadius": 30,
        "height": 150,
        "marginVertical": 24,
        "width": 150,
      }
    }
    testID="Lift Logo"
  />
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D3D3D3",
          "borderRadius": 25,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 20,
          "padding": 20,
          "width": "90%",
        }
      }
    >
      <Text />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Email..."
        rejectResponderTermination={true}
        style={
          Object {
            "flex": 1,
            "height": 50,
            "marginLeft": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D3D3D3",
          "borderRadius": 25,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 20,
          "padding": 20,
          "width": "90%",
        }
      }
    >
      <Text />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Password..."
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "flex": 1,
            "height": 50,
            "marginLeft": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#6200ee",
          "borderColor": "transparent",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 10,
            },
          ]
        }
        testID="Login Button"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "karla-bold",
                }
              }
            >
              LOGIN
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View>
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      visible={false}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text />
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#D3D3D3",
                  "borderRadius": 25,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 20,
                  "padding": 20,
                  "width": "90%",
                }
              }
            >
              <Text />
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                placeholder="Enter your email..."
                rejectResponderTermination={true}
                style={
                  Object {
                    "flex": 1,
                    "height": 50,
                    "marginLeft": 10,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#D3D3D3",
                  "borderRadius": 25,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 20,
                  "padding": 20,
                  "width": "90%",
                }
              }
            >
              <Text />
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                placeholder="Enter a username..."
                rejectResponderTermination={true}
                style={
                  Object {
                    "flex": 1,
                    "height": 50,
                    "marginLeft": 10,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#D3D3D3",
                  "borderRadius": 25,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 20,
                  "padding": 20,
                  "width": "90%",
                }
              }
            >
              <Text />
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                placeholder="Enter a password..."
                rejectResponderTermination={true}
                secureTextEntry={true}
                style={
                  Object {
                    "flex": 1,
                    "height": 50,
                    "marginLeft": 10,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#D3D3D3",
                  "borderRadius": 25,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "center",
                  "marginBottom": 20,
                  "padding": 20,
                  "width": "90%",
                }
              }
            >
              <Text />
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                placeholder="Confirm your password..."
                rejectResponderTermination={true}
                secureTextEntry={true}
                style={
                  Object {
                    "flex": 1,
                    "height": 50,
                    "marginLeft": 10,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <Text
              style={
                Object {
                  "marginLeft": 8,
                }
              }
            >
              I am a...
            </Text>
            <View>
              <RNCPicker
                items={
                  Array [
                    Object {
                      "label": "Coach",
                      "testID": undefined,
                      "textColor": undefined,
                      "value": "Coach",
                    },
                    Object {
                      "label": "Athlete",
                      "testID": undefined,
                      "textColor": undefined,
                      "value": "Athlete",
                    },
                  ]
                }
                onChange={[Function]}
                selectedIndex={0}
                style={
                  Array [
                    Object {
                      "height": 216,
                    },
                    undefined,
                  ]
                }
                testID="Role Picker"
              />
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#6200ee",
                  "borderColor": "transparent",
                  "borderRadius": 4,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "elevation": 2,
                  "marginVertical": 10,
                  "minWidth": 64,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 0.75,
                    "width": 0,
                  },
                  "shadowOpacity": 0.24,
                  "shadowRadius": 1.5,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": undefined,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 4,
                    },
                  ]
                }
                testID="Sign Up Button"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontWeight": "400",
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Array [
                          Object {
                            "letterSpacing": 1,
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          undefined,
                          Object {
                            "textTransform": "uppercase",
                          },
                          Object {
                            "color": "#ffffff",
                            "fontFamily": "System",
                            "fontWeight": "500",
                          },
                          Object {
                            "fontFamily": "System",
                            "fontWeight": "500",
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <Text
                      style={
                        Object {
                          "fontFamily": "karla-bold",
                        }
                      }
                    >
                      SIGNUP
                    </Text>
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginTop": 14,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "karla-bold",
          }
        }
      >
        SIGNUP
      </Text>
    </View>
  </View>
</View>
`;
